nifi-nar-bundles/nifi-box-bundle/nifi-box-services/src/main/java/org/apache/nifi/box/controllerservices/JsonConfigBasedBoxClientService.java

```diff
@@ -0,0 +1,162 @@
+        return PROPERTIES;
+    }
+
+    @OnEnabled
+    public void onEnabled(final ConfigurationContext context) {
+        boxAPIConnection = createBoxApiConnection(context);
+    }
+
+    @Override
+    protected Collection<ValidationResult> customValidate(ValidationContext validationContext) {
+        Collection<ValidationResult> validationResults = new ArrayList<>(super.customValidate(validationContext));
+
+        if (validationContext.getProperty(APP_CONFIG_FILE).isSet() && validationContext.getProperty(APP_CONFIG_JSON).isSet()) {
+            validationResults.add(new ValidationResult.Builder()
+                .subject("App configuration")
+                .valid(false)
+                .explanation(String.format("'%s' and '%s' cannot be configured at the same time",
+                    APP_CONFIG_FILE.getDisplayName(),
+                    APP_CONFIG_JSON.getDisplayName())
+                )
+                .build());
+        }
+
+        if (!validationContext.getProperty(APP_CONFIG_FILE).isSet() && !validationContext.getProperty(APP_CONFIG_JSON).isSet()) {
+            validationResults.add(new ValidationResult.Builder()
+                .subject("App configuration")
+                .valid(false)
+                .explanation(String.format("either '%s' or '%s' must be configured",
+                    APP_CONFIG_FILE.getDisplayName(),
+                    APP_CONFIG_JSON.getDisplayName())
+                )
+                .build());
+        }
+
+        return validationResults;
+    }
+
+    @Override
+    public BoxAPIConnection getBoxApiConnection() {
+        return boxAPIConnection;
+    }
+
+    private BoxAPIConnection createBoxApiConnection(ConfigurationContext context) {
+        BoxAPIConnection api;
+
+        String accountId = context.getProperty(ACCOUNT_ID).evaluateAttributeExpressions().getValue();
+        ProxyConfiguration proxyConfiguration = ProxyConfiguration.getConfiguration(context);
+
+        BoxConfig boxConfig;
+        if (context.getProperty(APP_CONFIG_FILE).isSet()) {
+            String appConfigFile = context.getProperty(APP_CONFIG_FILE).evaluateAttributeExpressions().getValue();
+            try (
+                Reader reader = new FileReader(appConfigFile);
+            ) {
+                boxConfig = BoxConfig.readFrom(reader);
+            } catch (FileNotFoundException e) {
+                throw new ProcessException("Couldn't find Box config file", e);
+            } catch (IOException e) {
+                throw new ProcessException("Couldn't read Box config file", e);
+            }
+        } else {
+            String appConfig = context.getProperty(APP_CONFIG_JSON).evaluateAttributeExpressions().getValue();
+            boxConfig = BoxConfig.readFrom(appConfig);
+        }
+
+        api = BoxDeveloperEditionAPIConnection.getAppEnterpriseConnection(boxConfig);
+        api.setProxy(proxyConfiguration.createProxy());
```

Comment:
The Proxy object which is set for the API does not contain username/password, when I tested the processor with username/password auth in the proxy, I got "HTTP/1.1 407 Proxy Authentication Required".
The username, password need to be passed to the Box API as well.

Reply:
@krisztina-zsihovszki @tpalfy I re-tested it with the updated code (setting proxy username and password) and it still does not work. I debugged it a bit and the issue seems to be in the Box API itself but it needs to be clarified.
I suggest removing the proxy property for now and going forward with merging the processors because they are valuable additions even without proxy support. The proxy issue can be investigated further in a follow-up jira.